opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java

```diff
@@ -108,7 +108,7 @@ public synchronized void setPluginDirectory(File pluginDirectory) {
      *
      * @param directory the directory path
      */
-    public void setPluginDirectory(String directory) {
+    public synchronized void setPluginDirectory(String directory) {
```

Comment:
Technically this is not needed. It silences Sonar and works because the lock used is reentrant.